src/Native/System.Security.Cryptography.Native/openssl.c

```diff
@@ -384,19 +384,29 @@ GetX509NameRawBytes(
         return 0;
     }
 
-// length is size_t on some platforms and int on others, so the comparisons
-// are not tautological everywhere. We can let the compiler optimize away
-// any part of the check that is.
-#pragma clang diagnostic push
-#pragma clang diagnostic ignored "-Wtautological-compare"
-    if (x509Name->bytes->length < 0 || x509Name->bytes->length > INT_MAX)
+    /* 
+     * length is size_t on some platforms and int on others, so the comparisons
+     * are not tautological everywhere. We can let the compiler optimize away
+     * any part of the check that is. We split the size checks into two checks
+     * so we can get around the warnings on Linux where the Length is unsigned
+     * whereas Length is signed on OS X. The first check makes sure the variable 
+     * value is less than INT_MAX in it's native format; once we know it is not
+     * too large, we can safely cast to an int to make sure it is not negative
+     */
+    if (x509Name->bytes->length > INT_MAX)
     {
-        assert(!"Huge/negative-length X509_NAME");
+        assert(0 && "Huge length X509_NAME");
```

Comment:
> This file is still .c so false won't work.

Will work whenever minimum requirement of building CoreFX would be set to VS2015:

* `bool` is ISO C99 type.
* VS2015 supports it.
* Moving forward, many C89 inabilities workaround can be removed in favor of VS2015's C99 support (or guarded with `#if defined(_MSC_VER) && _MSC_VER < 1900` for now, until VS2013 support is dropped).